delphi-frontend/src/test/java/au/com/integradev/delphi/reporting/QuickFixEditTest.java

```diff
@@ -0,0 +1,299 @@
+package au.com.integradev.delphi.reporting;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import au.com.integradev.delphi.DelphiProperties;
+import au.com.integradev.delphi.antlr.DelphiFileStream;
+import au.com.integradev.delphi.compiler.Platform;
+import au.com.integradev.delphi.core.Delphi;
+import au.com.integradev.delphi.file.DelphiFile.DelphiInputFile;
+import au.com.integradev.delphi.file.DelphiFileConfig;
+import au.com.integradev.delphi.preprocessor.DelphiPreprocessorFactory;
+import au.com.integradev.delphi.preprocessor.search.SearchPath;
+import au.com.integradev.delphi.reporting.edits.QuickFixEditImpl;
+import au.com.integradev.delphi.type.factory.TypeFactoryImpl;
+import au.com.integradev.delphi.utils.DelphiUtils;
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.Test;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.internal.TestInputFileBuilder;
+import org.sonar.plugins.communitydelphi.api.ast.DelphiAst;
+import org.sonar.plugins.communitydelphi.api.ast.DelphiNode;
+import org.sonar.plugins.communitydelphi.api.check.FilePosition;
+import org.sonar.plugins.communitydelphi.api.reporting.QuickFixEdit;
+
+class QuickFixEditTest {
+  private static final String TEST_ROOT = "/au/com/integradev/delphi/reporting";
+  private static final String TEST_UNIT_PATH =
+      "/au/com/integradev/delphi/reporting/TestQuickFixEditFile.pas";
+
+  private static DelphiFileStream getTestStream() {
+    try {
+      return new DelphiFileStream(
+          DelphiUtils.getResource(TEST_UNIT_PATH).getAbsolutePath(), "utf8");
```

Comment:
You can use the `StandardCharset` here and get the name instead of a magic string.